applications/matter_bridge/src/ble_providers/ble_bridged_device_factory.cpp

```diff
@@ -186,36 +191,32 @@ struct BluetoothConnectionContext {
 	bt_addr_le_t address;
 };
 
-void BluetoothDeviceConnected(bool discoverySucceeded, void *context)
+CHIP_ERROR BluetoothDeviceConnected(bool success, void *context)
 {
 	if (!context) {
-		return;
+		return CHIP_ERROR_INTERNAL;
 	}
 
 	BluetoothConnectionContext *ctx = reinterpret_cast<BluetoothConnectionContext *>(context);
 
-	VerifyOrExit(discoverySucceeded, );
-
-	/* Schedule adding device to main thread to avoid overflowing the BT stack. */
-	VerifyOrExit(chip::DeviceLayer::PlatformMgr().ScheduleWork(
-			     [](intptr_t context) {
-				     BluetoothConnectionContext *ctx =
-					     reinterpret_cast<BluetoothConnectionContext *>(context);
-				     chip::Optional<uint8_t> indexes[BridgeManager::kMaxBridgedDevicesPerProvider];
-				     chip::Optional<uint16_t> endpointIds[BridgeManager::kMaxBridgedDevicesPerProvider];
-				     if (CHIP_NO_ERROR != AddMatterDevices(ctx->deviceTypes, ctx->count, ctx->nodeLabel,
-									   ctx->provider)) {
-					     BLEConnectivityManager::Instance().RemoveBLEProvider(ctx->address);
-					     chip::Platform::Delete(ctx->provider);
-				     }
-				     chip::Platform::Delete(ctx);
-			     },
-			     reinterpret_cast<intptr_t>(ctx)) == CHIP_NO_ERROR, );
-	return;
+	if (!success) {
+		BLEConnectivityManager::Instance().RemoveBLEProvider(ctx->address);
```

Comment:
As we discussed, I think that it would be safer to move removing BLE provider to the BLE connectivity manager side.

Reply:
Yep, I will change that.